modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/api.mustache

```diff
@@ -0,0 +1,139 @@
+{{>licenseInfo}}
+@file:Suppress("UnusedImport")
+
+package {{apiPackage}}
+
+{{#imports}}import {{import}}
+{{/imports}}
+
+import {{packageName}}.infrastructure.*
+import io.ktor.client.HttpClient
+import io.ktor.client.request.forms.formData
+import io.ktor.client.engine.HttpClientEngine
+import io.ktor.client.features.json.serializer.KotlinxSerializer
+import io.ktor.client.request.forms.FormPart
+import io.ktor.client.utils.EmptyContent
+import kotlinx.serialization.json.Json
+import io.ktor.http.ParametersBuilder
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+{{#operations}}
+@Suppress(
+    "unused",
+    "RemoveRedundantQualifierName",
+    "RemoveExplicitTypeArguments",
+    "RemoveRedundantCallsOfConversionMethods",
+    "DuplicatedCode",
+    "UnnecessaryVariable",
+)
+public class {{classname}} : ApiClientBase {
+    public constructor(
+        baseUrl: String = "{{{basePath}}}",
+        httpClientEngine: HttpClientEngine? = null,
+        json: Json = Json {}
+    ) : super(baseUrl, httpClientEngine, json)
+
+    internal constructor(baseUrl: String, client: HttpClient, serializer: KotlinxSerializer) : super(
+        baseUrl,
+        client,
+        serializer
+    )
+
+    {{#operation}}
+    /**
+     * {{summary}}
+     * {{notes}}
+    {{#allParams}} * @param {{{paramName}}} {{description}} {{^required}}(optional{{#defaultValue}}, default to {{{.}}}{{/defaultValue}}){{/required}}
+    {{/allParams}} * @return {{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/returnType}}
+     */
+    {{#returnType}}
+    @Suppress("UNCHECKED_CAST")
+    {{/returnType}}
+    public suspend fun {{operationId}}(
+        {{#allParams}}
+        {{{paramName}}}: {{{dataType}}}{{^required}}? = null{{/required}}{{#hasMore}},{{/hasMore}}
+        {{/allParams}}
+    ): HttpResponse<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}Unit{{/returnType}}> {
+        {{! TODO: Optimize all the ?.let to only the places where it's needed }}
```

Comment:
Until optimized, let's place one global `@file:Suppress("UNNECESSARY_SAFE_CALL")` to the top of the file.
We have many of them and Kotlin compiler output is is flooded by these warnings.

Reply:
Yeah but this will probably be gone until I declare it ready anyway, so I didn't add it just yet

Reply:
Also I remember right now that this will be replaced with more "native" logic anyway if everything goes as planned, so that shouldn't be a problem. I just didn't have time yet to advance this

Reply:
This might also be fixed now, at least I don't see warnings generated in my client